### PR TITLE
NF: Added an option to include center feature in HollowSphere.

### PR DESCRIPTION
**Pull request by swaroopgj**
Added 'include_center' option HollowSphere to include center feature's data in the dataset sent to measures as first feature.

### CODE REVIEW

File: mvpa2/misc/neighborhood.py [hunk 1]
```diff
@@ -296,7 +299,7 @@ def _get_increments(self, ndim):
 
         if not len(res):
             warning("%s defines no neighbors" % self)
-        return res
+        return np.vstack([np.zeros(ndim,dtype='int'),res]) if self.include_center else res
```
Comment (yarikoptic) [inline mvpa2/misc/neighborhood.py]: you never assign it to self.  So it leads to tests failures:  https://travis-ci.org/PyMVPA/PyMVPA/jobs/47877969  and for proper this PR should have a test assigned (just set this parameter and verify that you do get center reported, and that it is first as you promised)

**Comment by swaroopgj** [comment]
Fixed it and added unit test.
Test passes locally.

File: mvpa2/misc/neighborhood.py [hunk 2]
```diff
@@ -253,6 +256,7 @@ def __init__(self, radius, inner_radius, **kwargs):
                   "than the radius (got %g)" % (inner_radius, radius)
         Sphere.__init__(self, radius, **kwargs)
         self._inner_radius = inner_radius
+        self.include_center = include_center
```
Comment (yarikoptic) [inline mvpa2/misc/neighborhood.py]: now also adjust ```__repr__``` below to add ```include_center ``` when it is not default... check other repr's for examples

**Comment by swaroopgj** [comment]
I added it to include_center to __repr__
Hopefully the right way :-|

File: mvpa2/misc/neighborhood.py [hunk 3]
```diff
@@ -253,10 +256,11 @@ def __init__(self, radius, inner_radius, **kwargs):
...
 
     def __repr__(self, prefixes=[]):
         return super(HollowSphere, self).__repr__(
-            ['inner_radius=%r' % (self._inner_radius,)])
+            ['inner_radius=%r' % (self._inner_radius,),'include_center=%r' % (self.include_center,)])
```
Comment (yarikoptic) [inline mvpa2/misc/neighborhood.py]: you did it almost right ;-)  so I will merge it and tune it up ... if you use

```Python
+ _repr_attrs(self, ['include_center'], default=False))
```
then it wouldn't even list if it it matches the default

**Pull request merged**